src/canister_tests/src/framework.rs

```diff
@@ -420,20 +425,35 @@ frame-ancestors 'none';$"
     .is_match(csp));
 }
 
-pub fn parse_metric(body: &str, metric: &str) -> (u64, SystemTime) {
+pub fn get_metrics(env: &StateMachine, canister_id: CanisterId) -> String {
+    let response = http_request(
+        env,
+        canister_id,
+        HttpRequest {
+            method: "GET".to_string(),
+            url: "/metrics".to_string(),
+            headers: vec![],
+            body: ByteBuf::new(),
+        },
+    )
+    .expect("HTTP request to /metrics failed");
+    String::from_utf8_lossy(&response.body).to_string()
+}
+
+pub fn parse_metric(body: &str, metric: &str) -> (f64, SystemTime) {
```

Comment:
There is some refactoring of test infrastructure (with regard to metrics) in here too. I could split that out, if you'd like.